// Ошибка в setTimeout

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script>
      // Что вы думаете? Выполнится ли .catch? Поясните свой ответ.

      new Promise(function (resolve, reject) {
        setTimeout(() => {
          throw new Error('Whoops!');
        }, 1000);
      }).catch(alert);

      // нет, потому что ошибка создается уже после выполнения кода
    </script>
  </body>
</html>
